react/components/RenderProvider.tsx

```diff
@@ -374,7 +374,18 @@ class RenderProvider extends Component<Props, RenderProviderState> {
     const transientRoute = {...route, ...navigationRoute}
     const {[page]: {allowConditions, declarer}} = pagesState
     const shouldSkipFetchNavigationData = !allowConditions && loadedPages.has(page)
-    const query = parse(location.search.substr(1))
+    const query = parse(window.location.search.substr(1))
+
+    // Store and pass disableUserLand logic to navigation
+    if(this.state.query && 'disableUserLand' in this.state.query && this.state.query.disableUserLand !== 'false'){
+      query['disableUserLand'] = this.state.query.disableUserLand
+      const queryString = '?' + Object.keys(query).map(key => `${key}${query[key]? '=' + query[key]: ''}`).join('&')
```

Comment:
Isn't there a native way to do this? Like URL()

Reply:
Don't think so, but I found we use https://github.com/ljharb/qs to do it. So the solution goes like this: `stringify(query)`